src/CharterManager.sol

```diff
@@ -258,7 +258,7 @@ contract CharterManagerImp {
     }
 
     function frob(bytes32 ilk, address u, address v, address w, int256 dink, int256 dart) external allowed(u) {
-        require(u == v && w == msg.sender, "CharterManager/not-matching");
+        require(u == v, "CharterManager/not-matching");
```

Comment:
I don't think this is safe anymore as user can repay DAI from any `w` address of their choosing without being authorized by the holder of the DAI. I would just copy this pattern from the `vat`:

```
// urn is either more safe, or the owner consents
require(either(both(dart <= 0, dink >= 0), wish(u, msg.sender)), "Vat/not-allowed-u");
// collateral src consents
require(either(dink <= 0, wish(v, msg.sender)), "Vat/not-allowed-v");
// debt dst consents
require(either(dart >= 0, wish(w, msg.sender)), "Vat/not-allowed-w");
```

I think we can axe the `u == v` requirement in this case as well.

Reply:
Right.
For fixing this I opted for using the allowed(w) modifier (similar to wish) and leaving the `u == v` restriction as well.
I think we can still be more restrictive as planned initially (and also try to minimize changes at this point).

Reply:
Hmm I think both are acceptable options but I tend to think mimicking the original `frob` `require`s is the perfect way to do it.
However if we want to keep it simpler for now the actual solution looks good as well, we can always upgrade the manager later.

Reply:
Discussed this with @gbalabasquer today.
We tend to leave the more restrictive conditions `u==v` and modifier `allowed(w)` here and not mimic the vat conditions. That might be done in another PR.
This is both since we don't want to put too many changes in this PR and since we see that in any case we will not make this contract a real vat passthrough. In this contract we won't add `fork` and also `move` has its delicacies (see other comments).